sql/parser/sql.y

```diff
@@ -1042,6 +1044,30 @@ show_stmt:
   {
     $$ = &ShowIndex{Table: $4}
   }
+| SHOW GRANTS on_privilege_target_clause for_grantee_clause
+  {
+		$$ = &ShowGrants{Targets: $3, Grantees: $4}
+  }
+
+on_privilege_target_clause:
+  ON privilege_target
+  {
+		$$ = &$2
```

Comment:
yup, that was it. I guess the generated code does something nasty to sqlDollar.
The diff shows going from "$$ = &($2)" to the tmp variables shows:
```
<                       sqlVAL.targetListPtr = &(sqlDollar[2].targetList)
---
>                       tmp := sqlDollar[2].targetList
>                       sqlVAL.targetListPtr = &tmp
```

I'm guessing [2].targetList gets zeroed, so a copy is needed.